contract_comm/currency/currency.go

```diff
@@ -118,7 +118,7 @@ func ConvertToGold(val *big.Int, currencyFrom *common.Address) (*big.Int, error)
 		return val, err
 	}
 
-	if currencyFrom == celoGoldAddress {
+	if *currencyFrom == celoGoldAddress {
```

Comment:
This definitely looks it was a bug.  But I don't think we have any code path that would hit that bug on the actual networks, because the places we call this function are only called with whitelisted fee currencies, and the Celo token contract isn't in the whitelist.  If it were added to the whitelist, we could hit this bug and the result would be that an error would be returned (due to divide by zero) and the transaction would not be added to the pool (but it would work if the transaction instead use feeCurrency: nil)

Reply:
totally... it was a bug ;)